translator/internaldata/oc_to_traces.go

```diff
@@ -150,16 +150,16 @@ func ocSpanToInternal(dest data.Span, src *octrace.Span) {
 	dest.SetDroppedEventsCount(droppedEventCount)
 	dest.SetLinks(links)
 	dest.SetDroppedLinksCount(droppedLinkCount)
-	ocStatusToInternal(src.Status, dest)
+	ocStatusToInternal(dest.Status(), src.Status)
 }
 
-func ocStatusToInternal(ocStatus *octrace.Status, out data.Span) {
+func ocStatusToInternal(dest data.SpanStatus, ocStatus *octrace.Status) {
 	if ocStatus == nil {
 		return
 	}
-	out.InitStatusIfNil()
-	out.Status().SetCode(data.StatusCode(ocStatus.Code))
-	out.Status().SetMessage(ocStatus.Message)
+	dest.InitEmpty()
```

Comment:
Just to clarify. `InitEmpty` is a mandatory call, right? Simply having an instance of SpanStatus does not guarantee that it is initialized and `SetX` functions may be called.

Reply:
Correct. Empty parents have nil embedded messages.

Reply:
Added comment in the generated code.